### PR TITLE
Dev.haa

### PR DESCRIPTION
**Pull request by shankhalika-katya**
Pull request template for adding a new language
-----------------------------------------------

*Do not use this for other types of pull requests*

* **Please check if the PR fulfills these requirements**
- [x] Mapping files are added in `g2p/mappings/langs`
- [x] Mapping is either added to an existing folder or a new folder has been added
- [x] Language folder and files use appropriate [ISO 639-3 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-3_codes)
- [x] Config.yaml file includes all author names, and settings necessary
- [x] Please add some test data in `g2p/tests/public/data`. The added file should be a csv/tsv/psv file and each row should have the format `[input_mapping_code,output_mapping_code,input_string,output_string]`
- [x] As the last step, G2P has been updated by running `g2p update` locally and committing the change
- [x] You agree to license your contribution under the same license as this project (see [LICENSE](https://github.com/roedoejet/g2p/blob/master/LICENSE) file).

* **Other information**:

### CODE REVIEW

File: g2p/tests/public/data/haa.csv [hunk 1]
```diff
@@ -0,0 +1,12 @@
+haa,haa-ipa,tth’ą̂ą̈̌,tθ'æ̃ɑ̃
+haa,haa-ipa,nhdhthtthddhwhshzhchch’yhkhghh'
+haa,haa-ipa,̥ndθtθʰtθ̥wʃʒtʃʰtʃ'jxɣhʔ
```
Comment (joanise) [inline g2p/tests/public/data/haa.csv:3]: These two test cases only have 3 fields each, there's a comma missing between the input and the reference.

**Review by joanise — CHANGES_REQUESTED** [review]
There was a merge conflict that `git merge` should have reported in `generated/config.yaml`, but instead it created an incorrect merge. I have fixed this.

There is also a problem with `tests/public/data/haa.csv`: two lines are missing a comma between the input and the reference, or maybe they're missing the reference, I'm not sure. Please check.

There is a third and more complicated problem: my tokenizer is compatible with two hops from lang to lang-ipa, but not yet with three hops from lang to lang-ipa. I'll have to fix that before the `haa.csv` test suite can pass completely.

**Comment by joanise** [comment]
@shankhalika-katya I'll let you fix the `haa.csv` file. You'll know it's fine when running `./test_langs.py` in `g2p/tests` succeeds and outputs no warnings. Please fetch my extra commit on your `dev.haa` branch first.

I'll fix the other issues that are blocking this PR.

The other two PRs are blocked by the same problems as this one, so I'll want to fix this PR first and then deal with the other two.

File: g2p/tests/public/data/haa.csv (hunk 1 above)
Comment (shankhalika-katya) [reply g2p/tests/public/data/haa.csv]: fixed!

**Review by joanise — APPROVED** [review]

**Comment by joanise** [comment]
Thanks @shankhalika-katya the fixed test suite. It works now.
Meanwhile, I have just fixed the tokenizer to support three hop settings like the one you needed for haa, so we can now merge this PR.

**Pull request merged**